oximeter/producer/src/lib.rs

```diff
@@ -208,57 +291,123 @@ impl Server {
             "address" => server.local_addr(),
             "interval" => ?server_info.interval,
         );
-        Ok(Self { registry, server })
+        Ok(Self { registry, registration_task, server })
+    }
+}
+
+/// Helper passed to the renewal task, used to determine whether / how to find
+/// Nexus periodically.
+enum FindNexus {
+    /// An explicit address was provided at creation time, just use it.
+    ByAddr(SocketAddr),
+    /// An address was not provided, we'll resolve it on each attempt to renew
+    /// the lease.
+    WithResolver(Resolver),
+}
+
+/// The default period within which we renew our lease from Nexus.
+const DEFAULT_LEASE_DURATION: Duration = Duration::from_secs(60 * 2);
```

Comment:
Do we need a default lease duration? This PR also changes Nexus to always return the lease duration, right?

Reply:
It does, yeah. I was trying to handle the case where we're registering with an older Nexus. That might have been possible during development, but it might be impossible now?

Reply:
I think it's impossible since the Nexus change landed first, yeah. The way we do updates it's not possible to have a newer service talk to a Nexus any older than the one that existed when it was built.

Reply:
Ok, I think that makes sense. I'll remove the const and propagate an error in the one place it's currently used. Thanks.

Reply:
Done in 1413ce8ed